app/imaging/svg.go

```diff
@@ -20,39 +20,67 @@ type SVGInfo struct {
 
 // ParseSVG returns information for the given SVG input data.
 func ParseSVG(svgReader io.Reader) (SVGInfo, error) {
-	var parsedSVG struct {
-		Width   string `xml:"width,attr,omitempty"`
-		Height  string `xml:"height,attr,omitempty"`
-		ViewBox string `xml:"viewBox,attr,omitempty"`
-	}
 	svgInfo := SVGInfo{
 		Width:  0,
 		Height: 0,
 	}
-	viewBoxPattern := regexp.MustCompile("^([0-9]+)[, ]+([0-9]+)[, ]+([0-9]+)[, ]+([0-9]+)$")
-	dimensionPattern := regexp.MustCompile("(?i)^([0-9]+)(?:px)?$")
 
-	// decode provided SVG
-	if err := xml.NewDecoder(svgReader).Decode(&parsedSVG); err != nil {
-		return svgInfo, err
-	}
+	decoder := xml.NewDecoder(svgReader)
 
-	// prefer viewbox for SVG dimensions over width/height
-	if viewBoxMatches := viewBoxPattern.FindStringSubmatch(parsedSVG.ViewBox); len(viewBoxMatches) == 5 {
-		svgInfo.Width, _ = strconv.Atoi(viewBoxMatches[3])
-		svgInfo.Height, _ = strconv.Atoi(viewBoxMatches[4])
-	} else if parsedSVG.Width != "" && parsedSVG.Height != "" {
-		widthMatches := dimensionPattern.FindStringSubmatch(parsedSVG.Width)
-		heightMatches := dimensionPattern.FindStringSubmatch(parsedSVG.Height)
-		if len(widthMatches) == 2 && len(heightMatches) == 2 {
-			svgInfo.Width, _ = strconv.Atoi(widthMatches[1])
-			svgInfo.Height, _ = strconv.Atoi(heightMatches[1])
+	for {
+		token, err := decoder.Token()
+		if err != nil {
+			return svgInfo, err
 		}
-	}
+		switch t := token.(type) {
+		case xml.StartElement:
+			for _, attr := range t.Attr {
+				if attr.Name.Local == "viewBox" {
+					values := strings.Fields(attr.Value)
+					if len(values) == 4 {
+						width := 0
+						_, widthErr := fmt.Sscan(values[2], &width)
+
+						height := 0
+						_, heightErr = fmt.Sscan(values[3], &height)
+
+						if widthErr != nil || heightErr != nil {
+							return svgInfo, err
+						}
+
+						svgInfo.Width = width
+						svgInfo.Height = height
+
+						return svgInfo, nil
+					}
+				}
+				if attr.Name.Local == "width" {
+					width := 0
+					_, err := fmt.Sscan(attr.Value, &width)
```

Comment:
I think that stuff that was removed was because the width/height properties could theoretically be `100px` instead of `100`. I don't know if that's actually the case, but maybe we can confirm that? Otherwise, it'll probably return an error here and hopefully not panic

Reply:
Oh, never mind. I didn't realize that using `fmt.Sscan` to parse the number just goes as long as it finds a number like the `parseInt` in JavaScript, so this should still work without the regex

![Screen Shot 2022-04-21 at 9 29 30 AM](https://user-images.githubusercontent.com/3277310/164468467-a15f3fc9-69db-441d-b6e7-867c4ccaff30.png)